web/src/main/java/org/openmrs/web/attribute/handler/ProgramFieldGenDatatypeHandler.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.attribute.handler;
+
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.Program;
+import org.openmrs.customdatatype.CustomDatatype;
+import org.openmrs.OpenmrsMetadata;
+import org.openmrs.customdatatype.datatype.BaseMetadataDatatype;
+import org.openmrs.customdatatype.CustomDatatype.Summary;
+import org.openmrs.customdatatype.InvalidCustomValueException;
+import org.openmrs.customdatatype.datatype.ProgramDatatype;
+import org.openmrs.serialization.SerializationException;
+import org.openmrs.serialization.SimpleXStreamSerializer;
+import org.springframework.stereotype.Component;
+
+/**
+ * Handler for the Program custom datatype
+ *
+ * @Component
+ * @since 1.12
+ */
+public class ProgramFieldGenDatatypeHandler<DT extends BaseMetadataDatatype<T>, T extends OpenmrsMetadata> extends BaseMetadataFieldGenDatatypeHandler<DT, T> {
```

Comment:
The first type parameter value should be **ProgramDatatype** and not **DT extends BaseMetadataDatatype<T>**